### PR TITLE
Commands/network TM

### PR DESCRIPTION
**Pull request by BurtBiel**
```
Command
    az network traffic-manager profile create

Arguments
    --name -n           [Required]: Name of resource.
    --resource-group -g [Required]: Name of resource group.
    --routing-method    [Required]: Allowed values: performance, priority, weighted.
    --unique-dns-name   [Required]: Relative DNS name for the traffic manager profile, resulting
                                    FQDN will be <uniqueDnsName>.trafficmanager.net, must be
                                    globally unique.
    --location -l                 : Location.  Default: global.
    --monitor-path                : Path to monitor.  Default: /.
    --monitor-port                : Port to monitor.  Default: 80.
    --monitor-protocol            : Default: http.
    --status                      : Create an enabled or disabled profile.  Default: enabled.
    --ttl                         : DNS Config time-to-live in seconds.  Default: 30.
```

```
Command
    az network traffic-manager profile check-dns

Arguments
    --name -n [Required]: DNS prefix to verify availability for.
```

```
Command
    az network traffic-manager endpoint create

Arguments
    --name -n           [Required]: Endpoint name.
    --profile-name      [Required]: Name of parent profile.
    --resource-group -g [Required]: Name of resource group.
    --type              [Required]: Endpoint type.  Values include: "azureEndpoints",
                                    "externalEndpoints", "nestedEndpoints".
    --endpoint-location           : Specifies the location of the external or nested endpoints when
                                    using the ΓÇÿPerformanceΓÇÖ traffic routing method.
    --endpoint-monitor-status     : Gets or sets the monitoring status of the endpoint.
    --endpoint-status             : Gets or sets the status of the endpoint..  If the endpoint is
                                    Enabled, it is probed for endpoint health and is included in the
                                    traffic routing method.  Possible values are 'Enabled' and
                                    'Disabled'.
    --min-child-endpoints         : Gets or sets the minimum number of endpoints that must be
                                    available in the child profile in order for the parent profile
                                    to be considered available. Only applicable to endpoint of type
                                    'NestedEndpoints'.
    --priority                    : Gets or sets the priority of this endpoint when using the
                                    ΓÇÿPriorityΓÇÖ traffic routing method. Possible values are from
                                    1 to 1000, lower values represent higher priority. This is an
                                    optional parameter.  If specified, it must be specified on all
                                    endpoints, and no two endpoints can share the same priority
                                    value.
    --target                      : Gets or sets the fully-qualified DNS name of the endpoint.
                                    Traffic Manager returns this value in DNS responses to direct
                                    traffic to this endpoint.
    --target-resource-id          : Gets or sets the Azure Resource URI of the of the endpoint.  Not
                                    applicable to endpoints of type 'ExternalEndpoints'.
    --weight                      : Gets or sets the weight of this endpoint when using the
                                    'Weighted' traffic routing method. Possible values are from 1 to
                                    1000.
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ adal==0.4.0
 applicationinsights==0.10.0
 argcomplete==1.3.0
 azure==2.0.0rc5
+azure-mgmt-trafficmanager==0.30.0rc5
```
Comment (tjprescott) [inline requirements.txt]: Is this not part of the normal SDK?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -360,3 +365,15 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 # VPN connection shared key
 register_cli_argument('network vpn-connection shared-key', 'connection_shared_key_name', CliArgumentType(options_list=('--name', '-n')), id_part='name')
 register_cli_argument('network vpn-connection shared-key', 'virtual_network_gateway_connection_name', CliArgumentType(options_list=('--connection-name',), metavar='NAME'), id_part='name')
+
+# Traffic manager profiles
+register_cli_argument('network traffic-manager profile', 'traffic_manager_profile_name', name_arg_type, id_part='name')
+register_cli_argument('network traffic-manager profile', 'profile_name', name_arg_type, id_part='name')
+register_cli_argument('network traffic-manager profile create', 'routing_method', choices=get_enum_choices(routingMethod))
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: This seems like it should have a completer (except for create).

File: requirements.txt (hunk 1 above)
Comment (BurtBiel) [inline requirements.txt]: nope, they have it separated out until it becomes stable (feature is in preview right now)

---
In reply to: [75039967](https://github.com/Azure/azure-cli/pull/708#discussion_r75039967) [](ancestors = 75039967)

**Comment by BurtBiel** [comment]
@tjprescott comments addressed

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 3]
```diff
@@ -360,3 +365,15 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 # VPN connection shared key
 register_cli_argument('network vpn-connection shared-key', 'connection_shared_key_name', CliArgumentType(options_list=('--name', '-n')), id_part='name')
 register_cli_argument('network vpn-connection shared-key', 'virtual_network_gateway_connection_name', CliArgumentType(options_list=('--connection-name',), metavar='NAME'), id_part='name')
+
+# Traffic manager profiles
+register_cli_argument('network traffic-manager profile', 'traffic_manager_profile_name', name_arg_type, id_part='name')
+register_cli_argument('network traffic-manager profile', 'profile_name', name_arg_type, id_part='name')
+register_cli_argument('network traffic-manager profile create', 'routing_method', choices=get_enum_choices(routingMethod), completer=get_enum_type_completion_list(routingMethod))
+register_cli_argument('network traffic-manager profile check-dns', 'name', name_arg_type, help='DNS prefix to verify availability for.', required=True)
+register_cli_argument('network traffic-manager profile check-dns', 'type', help=argparse.SUPPRESS, default='Microsoft.Network/trafficManagerProfiles')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: >etwork traffic-manager profile check-dns', 'type', h [](start = 24, length = 52)

Is there another value that could secretly be specified that would be valid, or does this exist simply for the template?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 4]
```diff
@@ -360,3 +365,15 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 # VPN connection shared key
 register_cli_argument('network vpn-connection shared-key', 'connection_shared_key_name', CliArgumentType(options_list=('--name', '-n')), id_part='name')
 register_cli_argument('network vpn-connection shared-key', 'virtual_network_gateway_connection_name', CliArgumentType(options_list=('--connection-name',), metavar='NAME'), id_part='name')
+
+# Traffic manager profiles
+register_cli_argument('network traffic-manager profile', 'traffic_manager_profile_name', name_arg_type, id_part='name')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: > profile', 'traffic_manager_ [](start = 46, length = 28)

Why no completer?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 5]
```diff
@@ -360,3 +365,15 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
...
+register_cli_argument('network traffic-manager profile', 'profile_name', name_arg_type, id_part='name')
+register_cli_argument('network traffic-manager profile create', 'routing_method', choices=get_enum_choices(routingMethod), completer=get_enum_type_completion_list(routingMethod))
+register_cli_argument('network traffic-manager profile check-dns', 'name', name_arg_type, help='DNS prefix to verify availability for.', required=True)
+register_cli_argument('network traffic-manager profile check-dns', 'type', help=argparse.SUPPRESS, default='Microsoft.Network/trafficManagerProfiles')
+
+# Traffic manager endpoints
+register_cli_argument('network traffic-manager endpoint', 'endpoint_name', name_arg_type, id_part='name', help='Endpoint name.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: >g_type, id_part='nam [](start = 82, length = 20)

Why no completer for endpoint_name?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py (hunk 3 above)
Comment (BurtBiel) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: not that I'm aware of.  I tried specifying another type and it told me that it had to be that type

---
In reply to: [75161529](https://github.com/Azure/azure-cli/pull/708#discussion_r75161529) [](ancestors = 75161529)

**Comment by BurtBiel** [comment]
@tjprescott comments addressed

---
In reply to: [240287751](https://github.com/Azure/azure-cli/pull/708#issuecomment-240287751) [](ancestors = 240287751)

**Comment by tjprescott** [comment]
:shipit:

**Pull request merged**